android/src/main/java/com/reactnativecommunity/viewpager/ReactViewPagerManager.java

```diff
@@ -1,132 +1,204 @@
+            @Override
+            public void onPageScrolled(int position, float positionOffset, int positionOffsetPixels) {
+                super.onPageScrolled(position, positionOffset, positionOffsetPixels);
+                eventDispatcher.dispatchEvent(
+                        new PageScrollEvent(vp.getId(), position, positionOffset));
+            }
+
+            @Override
+            public void onPageSelected(int position) {
+                super.onPageSelected(position);
+                eventDispatcher.dispatchEvent(
+                        new PageSelectedEvent(vp.getId(), position));
+            }
+
+            @Override
+            public void onPageScrollStateChanged(int state) {
+                super.onPageScrollStateChanged(state);
+                String pageScrollState;
+                switch (state) {
+                    case SCROLL_STATE_IDLE:
+                        pageScrollState = "idle";
+                        break;
+                    case SCROLL_STATE_DRAGGING:
+                        pageScrollState = "dragging";
+                        break;
+                    case SCROLL_STATE_SETTLING:
+                        pageScrollState = "settling";
+                        break;
+                    default:
+                        throw new IllegalStateException("Unsupported pageScrollState");
+                }
+                eventDispatcher.dispatchEvent(
+                        new PageScrollStateChangedEvent(vp.getId(), pageScrollState));
+            }
+        });
+        return vp;
+    }
+
+    @Override
+    public void addView(ViewPager2 parent, View child, int index) {
+        if (child == null) {
+            return;
+        }
+        reactChildrenViews.put(child.getId(), child);
+        ((FragmentAdapter) parent.getAdapter()).addFragment(child, index);
+    }
+
+    @Override
+    public int getChildCount(ViewPager2 parent) {
+        return parent.getAdapter().getItemCount();
+    }
+
+
+    @Override
+    public View getChildAt(ViewPager2 parent, int index) {
+        return ((FragmentAdapter) parent.getAdapter()).getChildAt(index);
+    }
+
+    @Override
+    public void removeView(ViewPager2 parent, View view) {
+        reactChildrenViews.remove(view.getId());
+        ((FragmentAdapter) parent.getAdapter()).removeFragment(view);
+        super.removeView(parent, view);
+
+    }
+
+    @Override
+    public boolean needsCustomLayoutForChildren() {
+        return true;
+    }
+
+
+    @ReactProp(name = "scrollEnabled", defaultBoolean = true)
+    public void setScrollEnabled(ViewPager2 viewPager, boolean value) {
+        viewPager.setUserInputEnabled(value);
+    }
+
+    @ReactProp(name = "orientation")
+    public void setOrientation(ViewPager2 viewPager, String value) {
+        viewPager.setOrientation(value.equals("horizontal") ? ORIENTATION_HORIZONTAL : ViewPager2.ORIENTATION_VERTICAL);
```

Comment:
I would change it to:

```
viewPager.setOrientation(value.equals("vertical") ? ORIENTATION_VERTICAL : ViewPager2. ORIENTATION_HORIZONTAL);
```

So by default is `horizontal` as it has always been.